pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -0,0 +1,195 @@
+	connTrackAcct, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_acct")
+	if err != nil {
+		if !os.IsPermission(err) {
+			klog.Errorf("Error when getting net.netfilter.nf_conntrack_acct")
+			return nil, err
+		} else {
+			klog.Errorf("Permission denied to access net.netfilter.nf_conntrack_acct: Counters in flow records may not update")
+		}
+	} else {
+		if connTrackAcct == 0 {
+			err = sysctl.SetSysctlNet("netfilter/nf_conntrack_acct", 1)
+			if err != nil {
+				if !os.IsPermission(err) {
+					klog.Errorf("Error when setting net.netfilter.nf_conntrack_acct")
+					return nil, err
+				} else {
+					klog.Errorf("Permission denied to access net.netfilter.nf_conntrack_acct: Counters in flow records may not update")
+				}
+			}
+		}
+	}
+
+	// ZoneID filter is not supported currently in tl-mo/conntrack library.
+	// Link to issue: https://github.com/ti-mo/conntrack/issues/23
+	// Dump all flows in the conntrack table for now.
+	conns, err := ct.connTrackPoller.DumpFilter(conntrack.Filter{})
+	if err != nil {
+		klog.Errorf("Error when dumping flows from conntrack: %v", err)
+		return nil, err
+	}
+
+	filteredConns := make([]*flowexporter.Connection, 0, len(conns))
+	for _, conn := range conns {
+		if conn.Zone != openflow.CtZone {
+			continue
+		}
+		srcIP := conn.TupleOrig.IP.SourceAddress
+		dstIP := conn.TupleReply.IP.SourceAddress
+		// Only get Pod-to-Pod flows. Pod-to-ExternalService flows are ignored for now.
+		if srcIP.Equal(ct.nodeConfig.GatewayConfig.IP) || dstIP.Equal(ct.nodeConfig.GatewayConfig.IP) {
+			continue
+		}
+
+		// Pod-to-Service flows w/ kube-proxy: There are two conntrack flows for every Pod-to-Service flow.
+		// One is with ClusterIP as source or destination, where other IP is podIP. Second conntrack flow is
+		// with resolved Endpoint Pod IP corresponding to ClusterIP. Both conntrack flows have same stats, which makes them duplicate.
+		// Ideally, we have to correlate these two connections and maintain one connection with both Endpoint Pod IP and ClusterIP.
+		// To do the correlation, we need ClusterIP-to-EndpointIP mapping info, which is not available at Agent.
+		// Therefore, we ignore the connection with ClusterIP and keep the connection with Endpoint Pod IP.
+		// Conntrack flows will be different for Pod-to-Service flows w/ Antrea-proxy. This implementation will be simpler, when the
+		// Antrea proxy is supported.
+		if ct.serviceCIDR.Contains(srcIP) || ct.serviceCIDR.Contains(dstIP) {
+			continue
+		}
+		filteredConns = append(filteredConns, createAntreaConn(&conn))
+	}
+
+	klog.V(2).Infof("Finished poll cycle -- total flows: %d flows in Antrea zoneID: %d", len(conns), len(filteredConns))
+
+	return filteredConns, nil
+}
+
+// ConnTrackPoller is an interface created to consume the required functions from the third party
+// conntrack library. This is helpful in writing unit tests.
+var _ ConnTrackPoller = new(connTrackPoller)
+
+type ConnTrackPoller interface {
```

Comment:
I think I have left a similar comment before but I am still confused about the `Poller` name given that there is no polling happening in this interface or its implementation. Why not `ConnTrackDumper` since the main method is `DumpFilter`?

Reply:
Although `ConnTrackDumper` would also be a good name for the current `ConnTrack` interface as per go conventions, so maybe interfaces should be renamed as follows:
 * `ConnTrack` -> `ConnTrackDumper`
 * `ConnTrackPoller` -> `ConnTrackProxier` (since it proxies requests to the third party library?)

Reply:
I was going for poller for the object which is getting conntrack flows from the module, but I see what you are saying `poll()` of `ConnectionStore`, which is invoking the call periodically. 
`ConnTrackDumper` sounds good. In the future patch (shorturl.at/tCFGW), `ConnTrackPoller`is implemented by 'ConnTrackSystem' and 'ConnTrackNetdev' for kernel and userspace dataplanes. As userspace dataplane do not require third party library, can we just call it as ConnTrack?

Reply:
Maybe just `ConnTrack` does not convey enough information. What about `ConnTrackInterface` or `ConnTrackInterfacer`. There isn't really a "strict" convention in Go, except for interfaces with a single method.

Reply:
Thought of ConnTrackInterface.. felt it is redundant with datatype. Liked `ConnTrackInterfacer`. Sticking with it. Thanks.